src/sentry/tasks/weekly_reports.py

```diff
@@ -0,0 +1,803 @@
+from sentry.utils import json
+from sentry.utils.dates import floor_to_utc_day, to_datetime, to_timestamp
+from sentry.utils.email import MessageBuilder
+from sentry.utils.outcomes import Outcome
+from sentry.utils.query import RangeQuerySetWrapper
+from sentry.utils.snuba import parse_snuba_datetime, raw_snql_query
+
+ONE_DAY = int(timedelta(days=1).total_seconds())
+date_format = partial(dateformat.format, format_string="F jS, Y")
+
+
+class OrganizationReportContext:
+    def __init__(self, timestamp, duration, organization):
+        self.timestamp = timestamp
+        self.duration = duration
+
+        self.start = to_datetime(timestamp - duration)
+        self.end = to_datetime(timestamp)
+
+        self.organization = organization
+        self.projects = {}  # { project_id: ProjectContext }
+
+        self.project_ownership = {}  # { user_id: set<project_id> }
+        for project in organization.project_set.all():
+            self.projects[project.id] = ProjectContext(project)
+
+    def __repr__(self):
+        return self.projects.__repr__()
+
+
+class ProjectContext:
+    accepted_error_count = 0
+    dropped_error_count = 0
+    accepted_transaction_count = 0
+    dropped_transaction_count = 0
+
+    all_issue_count = 0
+    existing_issue_count = 0
+    reopened_issue_count = 0
+    new_issue_count = 0
+
+    def __init__(self, project):
+        self.project = project
+
+        # Array of (group_id, group_history, count)
+        self.key_errors = []
+        # Array of (transaction_name, count_this_week, p95_this_week, count_last_week, p95_last_week)
+        self.key_transactions = []
+        # Array of (Group, count)
+        self.key_performance_issues = []
+
+        # Dictionary of { timestamp: count }
+        self.error_count_by_day = {}
+        # Dictionary of { timestamp: count }
+        self.transaction_count_by_day = {}
+
+    def __repr__(self):
+        return f"{self.key_errors}, Errors: [Accepted {self.accepted_error_count}, Dropped {self.dropped_error_count}]\nTransactions: [Accepted {self.accepted_transaction_count} Dropped {self.dropped_transaction_count}]"
+
+
+# The entry point. This task is scheduled to run every week.
+@instrumented_task(
+    name="sentry.tasks.weekly_reports.schedule_organizations",
+    queue="reports.prepare",
+    max_retries=5,
+    acks_late=True,
+)
+def schedule_organizations(dry_run=False, timestamp=None, duration=None):
+    if timestamp is None:
+        # The time that the report was generated
+        timestamp = to_timestamp(floor_to_utc_day(timezone.now()))
+
+    if duration is None:
+        # The total timespan that the task covers
+        duration = ONE_DAY * 7
+
+    organizations = Organization.objects.filter(status=OrganizationStatus.ACTIVE)
+    for i, organization in enumerate(
+        RangeQuerySetWrapper(organizations, step=10000, result_value_getter=lambda item: item.id)
+    ):
+        if features.has("organizations:weekly-email-refresh", organization):
```

Comment:
We have not yet enabled this on any org; correct?

Reply:
Nope. The feature flag doesn't exist yet. The plan is to start by enabling it for Sentry.